contracts/margined_engine/src/query.rs

```diff
@@ -70,12 +71,14 @@ pub fn query_all_positions(deps: Deps, trader: String) -> StdResult<Vec<Position
     };
 
     for vamm in vamms.iter() {
-        let position_key = keccak_256(&[vamm.as_bytes(), trader.as_bytes()].concat());
-        let position = read_position(deps.storage, &position_key)?;
-
-        // a default is returned if no position found with no trader set
-        if position.trader == trader {
-            response.push(position)
+        let position_key = keccak_256(&[vamm.as_bytes()].concat());
+        let positions = read_positions(deps.storage, &position_key, start_after, limit, order_by).unwrap();
```

Comment:
should have filter fn to remain paging

Reply:
resolve in https://github.com/oraichain/perpetuals/pull/1/commits/40b7f53165f2c6446eb09440673080d83d0d69b0